lib/Doctrine/Helpers/NamingHelper.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Helpers;
+
+use Throwable;
+use function in_array;
+use function preg_match;
+
+final class NamingHelper
+{
+    public static function hasExceptionSuffix(string $className) : bool
+    {
+        return preg_match('~Exception$~', $className) === 1;
```

Comment:
Interfaces have to end with `Exception`. I can add `ExceptionInterface` to a test for a fail case.

Reply:
> Interfaces have to end with Exception

That's true only for doctrine code. Userland (or doctrine codebase itself) can be extending 3rd party exceptions with different naming convention. Detecting Exception(Interface)? is small step but easy win in such cases

Reply:
True. It started for Doctrine code, but if someone else wants to use it, this might become a nuisance. But the detection of what is an Exception and what not depends on a naming convention. Your example uses a classic "end with Interface" convention, but what about a 3rd party exception naming convention without any suffixes or prefixes? This would make detecting exceptions with this sniff very hard.

What about making the naming convention configurable?

Reply:
If you want to make this configurable, feel free. For me, handling ExceptionInterface would be good enough step, as such convention is common.